### PR TITLE
Product filter Added with few more bug fix

### PR DESCRIPTION
**Pull request by animeshgarai09**
Feature added in this PR

- Filter added for shop page
- FIlter with search parameter added
- Countdown added for deals of the day

### CODE REVIEW

File: src/pages/home/sub-components/countdown.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import styles from "../home.module.scss"
+import useCountDown from "@hooks/useCountDown"
+
+const Countdown = ({ timer }) => {
+    const { days, hours, minutes, seconds } = useCountDown(timer)
```
Comment (vashnavichauhan1825) [inline src/pages/home/sub-components/countdown.js:5]: appreciation:- like you separated the code  of functionality

File: src/pages/shop/shop.js [hunk 2]
```diff
@@ -1,83 +1,62 @@
...
-                                </ProductActions>
-                            </ProductBody>
-                        </ProductWrapper>
-                    )
-                })}
+                {
+                    filter && filteredProducts.length === 0 && <h1>Not found 404</h1>
+                }
+                {!filter && filteredProducts.length === 0
+                    ? [...Array(10)].map((_, i) => {
```
Comment (vashnavichauhan1825) [inline src/pages/shop/shop.js:31]: here learned something new about the throwaway variable thanks!

**Review by vashnavichauhan1825 — APPROVED** [review]

**Comment by vashnavichauhan1825** [comment]
code looks clean and readable 🙌 ready to merge

**Review by arjun-computer-geek — APPROVED** [review]
good code of conduct. Great job

**Comment by yash9263** [comment]
UI looks great. code looks clean and clear nice use of util functions, custom hooks, and reusable components.

**Review by yash9263 — APPROVED** [review]

**Review by aayushjain-code — APPROVED** [review]
The code looks clean and Very nice UI ready to merge.

**Pull request merged**